docs/src/main/asciidoc/writing-extensions.adoc

```diff
@@ -1290,6 +1290,143 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
 }
 ----
 
+=== Extension Metrics
+
+An extension can decide to provide metrics through the `quarkus-smallrye-metrics` extension.
+A typical use case for this would be that an extension scans the application code for relevant components (like entities, messaging endpoints, etc.)
+and creates a set of metrics for each of these components. There is a unified mechanism for metric registration - it uses the `MetricBuildItem` class
+provided by the `quarkus-smallrye-metrics-spi` module.
+
+There are several distinct situations that can happen and each requires a bit different handling:
+
+1. The underlying library used by your extension is using MP Metrics API directly.
```

Comment:
```suggestion
1. The underlying library used by your extension is using the MicroProfile Metrics API directly.
```
(for the first occurrence, this might not be obvious to the reader)